File: DragonShell/test-files/subprocess.py
import os
import time

def main():
    # Print message indicating subprocess is running
    print("Running subprocess")

    # Open a file to write factors to
    try:
        with open("test-files/factors.txt", "w") as file:
            # Loop through numbers to find factors
            for i in range(1, 100000):  # Start from 1 to avoid division by zero
                for j in range(1, i + 1):  # Change condition to j <= i for valid factors
                    if i % j == 0:
                        file.write(f"Found a factor of i for i = {i}, j = {j}\n")
                        file.flush()  # Ensure the output is written immediately

                        # Optionally, sleep to simulate longer processing time
                        # time.sleep(0.0001)  # Sleep for 100 microseconds
    except IOError as e:
        print(f"Failed to open file: {e}")
        return os.EX_OSFILE  # Return appropriate error code

    return 0

if __name__ == "__main__":
    main()

File: DragonShell/test-files/subprocess.c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>

// simulates user and sys time of about 27 and 3 seconds respectivly for 200000 iterations and 6 and 1 for 100000 iterations 

int main() {
    // Set up signal handlers

    printf("Running subprocess\n");

    // Open a file to write factors to
    FILE *file = fopen("test-files/factors.txt", "w");
    if (file == NULL) {
        perror("Failed to open file");
        return EXIT_FAILURE;
    }

    // for (int i = 1; i < 200000; i++) { // Start from 1 to avoid division by zero
    for (int i = 1; i < 100000; i++) { // Start from 1 to avoid division by zero
        for (int j = 1; j <= i; j++) { // Change condition to j <= i for valid factors
            if (i % j == 0) {
                fprintf(file, "Found a factor of i for i = %d, j = %d\n", i, j);
                fflush(file);  // Ensure the output is written immediately

                // Optionally, sleep to simulate longer processing time
                // usleep(100);  // Sleep for 100 microseconds
            }
        }
    }

    fclose(file); // Close the file to flush and release resources

    return 0;
}